src/template/Template.js

```diff
@@ -34,96 +33,83 @@ export const Template = class {
   }
 
   cacheKey(prompt, { systemPrompt, format, cacheHint, schema }) {
-    const hash = CryptoJS.SHA256(JSON.stringify({ prompt, systemPrompt, format, cacheHint, schema }))
-      .toString(CryptoJS.enc.Hex)
-      .substr(0, 16);
+    const hash = hashObjectShort({ prompt, systemPrompt, format, cacheHint, schema });
     const promptPart = prompt.replaceAll(/[^A-Za-z0-9]+/g, '-').substr(0, 32);
     return `ai-${this.constructor.name}-${this.model}-${promptPart}-${hash}`;
   }
 
-  async renderMulti(context, flexField, ai, cache) {
-    const copy = { ...context };
+  async renderMulti(context, flexField, ai, cache, timer) {
```

Comment:
instead of passing `timer`, can we integrate it into logger somehow, or at least make it a globally imported constant like logger?

so we don't need to always pass it around everywhere